src/core/packages/http/server-internal/src/cookie_session_storage.ts

```diff
@@ -30,21 +32,41 @@ class ScopedCookieSessionStorage<T extends object> implements SessionStorage<T>
   public async get(): Promise<T | null> {
     try {
       const session = await this.server.auth.test('security-cookie', this.request);
+
       // A browser can send several cookies, if it's not an array, just return the session value
-      if (!Array.isArray(session)) {
+      if (!Array.isArray(session.credentials)) {
         return session.credentials as T;
       }
 
       // If we have an array with one value, we're good also
-      if (session.length === 1) {
-        return session[0] as T;
+      if (session.credentials.length === 1) {
+        return session.credentials[0] as T;
+      }
+
+      // If we have more than one session, return the first one if they are all the same
+      if (session.credentials.length > 1) {
+        this.log.warn(
+          `Found multiple auth sessions. Found:[${session.credentials.length}] sessions. Checking equality...`
+        );
+        const [firstSession] = session.credentials;
+        const allEqual = session.credentials.every((s) => {
```

Comment:
nit: we're checking the first session against itself, we might skip one iteration with the following:
```suggestion
        const [firstSession, ...rest] = session.credentials;
        const allEqual = rest.every((s) => {
```

Reply:
IMO, the suggested change is a change request